File: solutions/c/darts/1/darts.c
// Dedicated to Shree DR.MDD

#include "darts.h"
#include <math.h>

uint8_t score(coordinate_t point) {
    float dist = hypot(point.x, point.y);
    if (dist <= 1.0F) {
            return 10;
        }
    if (dist <= 5.0F) {
            return 5;
        }
    if (dist <= 10.0F) {
            return 1;
        }
    return 0;
}

File: solutions/r/darts/1/darts.R
score <- function(x, y) {
  distance_squared <- x^2 + y^2
  
  if (distance_squared <= 1^2) {
    10
  } else if (distance_squared <= 5^2) {
    5
  } else if (distance_squared <= 10^2) {
    1
  } else {
    0
  }
}